symfit/symmip/mip.py

```diff
@@ -0,0 +1,197 @@
+from dataclasses import dataclass, field
+from typing import List, Dict, Protocol
+from functools import cached_property, reduce
+from collections import defaultdict, namedtuple
+import operator
+import itertools
+
+from sympy import Expr, Rel, Eq, Idx, Indexed, Symbol, lambdify
+from sympy.printing.pycode import PythonCodePrinter
+import numpy as np
+
+from symfit.symmip.backend import GurobiBackend
+from symfit.core.fit_results import FitResults
+from symfit.core.support import key2str
+
+
+DummyModel = namedtuple('DummyModel', 'params')
+
+VTYPES = {'binary': 'B', 'integer': 'I', 'real': 'C',} # ToDo: Enum?
+
+
+class MIPBackend(Protocol):
```

Comment:
Should the GurobiBackend be a subclass of this?

Reply:
For now I went with a protocol because it allows for static typing and duck-typing. An ABC was not worth it (yet) because there was no code reuse. But maybe I will make a common base class now that we have two solvers, but I'll keep the protocol nonetheless.